### DISCUSSION TITLE
Error when executing ros2 launch

### DISCUSSION THREAD

**Discussion by vvitzilaios** [gstavrinos/ai_msc_robotic_systems_assignments#9] [Q&A] [answered]
After executing `ros2 launch tt_umpire sim_tt.launch.py assignment:=1 target_node_name:=tt_ball_locator` I receive the following:

`[ERROR] [launch]: Caught exception in launch (see debug for traceback): "package 'gazebo_ros' not found, searching: ['/media/psf/ros2_ws/install/urdf_test', '/media/psf/ros2_ws/install/tt_umpire', '/media/psf/ros2_ws/install/tt_table_explorer', '/media/psf/ros2_ws/install/tt_racket_handling', '/media/psf/ros2_ws/install/tt_ball_locator', '/media/psf/ros2_ws/install/tiago_simulation', '/media/psf/ros2_ws/install/gazebo_table_tennis', '/media/psf/ros2_ws/install/tiago_robot', '/media/psf/ros2_ws/install/tiago_navigation', '/media/psf/ros2_ws/install/tiago_gazebo', '/media/psf/ros2_ws/install/tiago_moveit_config', '/media/psf/ros2_ws/install/tiago_bringup', '/media/psf/ros2_ws/install/tiago_description', '/media/psf/ros2_ws/install/tiago_controller_configuration', '/media/psf/ros2_ws/install/tiago_2dnav', '/media/psf/ros2_ws/install/pmb2_robot', '/media/psf/ros2_ws/install/pmb2_navigation', '/media/psf/ros2_ws/install/pmb2_2dnav', '/media/psf/ros2_ws/install/pmb2_maps', '/media/psf/ros2_ws/install/pmb2_bringup', '/media/psf/ros2_ws/install/pmb2_description', '/media/psf/ros2_ws/install/pmb2_controller_configuration', '/media/psf/ros2_ws/install/pal_gripper', '/media/psf/ros2_ws/install/pal_gripper_description', '/media/psf/ros2_ws/install/pal_gripper_controller_configuration', '/media/psf/ros2_ws/install/pal_gazebo_worlds', '/media/psf/ros2_ws/install/launch_pal', '/media/psf/ros2_ws/install/hey5_description', '/media/psf/ros2_ws/install/diff_drive_controller', '/opt/ros/humble']"`

Also:
- Should I install gazebo separately? 
- Should I install rviz separately?
- There might be a potential issue with gazebo and arm64 kernels.
- I want to kms...

**Comment by gstavrinos** [comment]
Provide me with the output of `rosdep install --from-paths src -ri -y` and `sudo apt install ros-humble-gazebo-ros`.

As you probably already know, the rosdep command should be run from the root of your workspace.

**Comment by vvitzilaios** [comment]
- `rosdep install --from-paths src -ri -y`

```
parallels@ubuntu-linux-22-04-desktop:/media/psf/ros2_ws$ rosdep install --from-paths src -ri -y
executing command [sudo -H apt-get install -y ros-humble-gazebo-plugins]
[sudo] password for parallels: 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package ros-humble-gazebo-plugins
executing command [sudo -H apt-get install -y ros-humble-gazebo-ros]
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package ros-humble-gazebo-ros
executing command [sudo -H apt-get install -y ros-humble-gazebo-ros2-control]
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package ros-humble-gazebo-ros2-control
executing command [sudo -H apt-get install -y ros-humble-pal-gazebo-plugins]
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package ros-humble-pal-gazebo-plugins
executing command [sudo -H apt-get install -y ros-humble-nav2-bringup]
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 krb5-multidev : Depends: libkrb5-3 (= 1.19.2-2ubuntu0.1) but 1.19.2-2ubuntu0.2 is to be installed
                 Depends: libk5crypto3 (= 1.19.2-2ubuntu0.1) but 1.19.2-2ubuntu0.2 is to be installed
                 Depends: libgssapi-krb5-2 (= 1.19.2-2ubuntu0.1) but 1.19.2-2ubuntu0.2 is to be installed
E: Unable to correct problems, you have held broken packages.
ERROR: the following rosdeps failed to install
  apt: command [sudo -H apt-get install -y ros-humble-gazebo-plugins] failed
  apt: command [sudo -H apt-get install -y ros-humble-gazebo-ros] failed
  apt: command [sudo -H apt-get install -y ros-humble-gazebo-ros2-control] failed
  apt: command [sudo -H apt-get install -y ros-humble-pal-gazebo-plugins] failed
  apt: command [sudo -H apt-get install -y ros-humble-nav2-bringup] failed
  apt: Failed to detect successful installation of [ros-humble-gazebo-plugins]
  apt: Failed to detect successful installation of [ros-humble-gazebo-ros]
  apt: Failed to detect successful installation of [ros-humble-gazebo-ros2-control]
  apt: Failed to detect successful installation of [ros-humble-pal-gazebo-plugins]
  apt: Failed to detect successful installation of [ros-humble-nav2-bringup]
```

- `sudo apt install ros-humble-gazebo-ros`

```
parallels@ubuntu-linux-22-04-desktop:/media/psf/ros2_ws$ sudo apt install ros-humble-gazebo-ros
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package ros-humble-gazebo-ros
```

Should I set the repository to a different one? Is this why apt can't find the packages of gazebo?

**Reply by vvitzilaios** [reply, marked as answer]
<img width="537" alt="image" src="https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/118114473/ac83f52e-ce13-4e55-a4b0-42188132f3b8">

I have arm64 architecture
